### PR TITLE
[luci/pass] Introduce RemoveUnnecessaryReshapePass on luci/pass

### PR DESCRIPTION
**Pull request by struss**
For #5575 

This commit introduce a `RemoveUnnecessaryReshapePass` on luci/pass.

ONE-DCO-1.0-Signed-off-by: KiDeuk Bang <rrstrous@nate.com>

### CODE REVIEW

File: compiler/luci/pass/src/RemoveUnnecessaryReshapePass.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  std::vector<int32_t> shape_info;
+  auto new_shape = dynamic_cast<luci::CircleConst *>(target_node->shape());
+  // If shape node is not a const node, get updated shape info from option.
+  if (new_shape == nullptr)
+  {
+    auto dummy_shape = dynamic_cast<luci::CircleOutputDummy *>(target_node->shape());
+    if (dummy_shape == nullptr)
+      return false;
+
+    // Need to extract new_shape info on target_node->newShape()
```
Comment (mhs4670go) [inline compiler/luci/pass/src/RemoveUnnecessaryReshapePass.cpp:39]: @struss 
AFAIK, `CircleOutputDummy` is inserted when `Reshape` op doesn't have its option, which means it doesn't have `newShape` attribute as well. This line can be run?

Comment (struss) [reply compiler/luci/pass/src/RemoveUnnecessaryReshapePass.cpp:39]: @mhs4670go sorry for my misunderstanding. I think your right. from `luci/import` logic, if second input not comes, shape is created from option. so if `CircleOutputDummy` comes in shape, it have neither shape and reshape option. 
I will fix this matching with `luci/import` logic. Thanks for pointing out!

Comment (struss) [reply compiler/luci/pass/src/RemoveUnnecessaryReshapePass.cpp]: and also, if shape is `CircleOutputDummy`, then options also no value, follow from `luci/import` (can't extract shape from option)
https://github.com/Samsung/ONE/blob/5b4a868d686fe4d19af0c99921f784de8e027504/compiler/luci/import/src/Nodes/CircleReshape.cpp#L68-L76

**Review by mhs4670go — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**